aiconfig-docs/docs/gradio-workbook.md

```diff
@@ -0,0 +1,144 @@
+
+**[Gradio Workbooks](https://huggingface.co/spaces/lastmileai/gradio-workbook-template) provide an out-of-the-box playground for creating Hugging Face Spaces.**
+
+The playground provides instant model selection, chaining of models, and templates - no setup or code required!
+
+<div align="center">
+  <picture>
+    <img alt="workbook" src="https://github.com/lastmile-ai/aiconfig/assets/81494782/494523e1-23c6-4709-94cb-7ebe86873dfd" width="600"/>
+  </picture>
+</div>
+<br/>\
+Gradio Workbook unlocks new capabilities to HuggingFace Spaces users!
+
+1. **Build apps inspired by Spaces**. Gradio Workbook spaces are backed by an `aiconfig.json` file that stores prompts and model settings. Click **Download** to get the aiconfig file for your space. Use the `aiconfig.json` in your application code with the [AIConfig SDK](https://github.com/lastmile-ai/aiconfig).
+2. **Share read-only views of Spaces**. You can now share your workflow, prompts, and outputs, even as a space viewer. Check out this [space](https://huggingface.co/spaces/lastmileai/music_generator) and it's [read-only view](https://lastmileai.dev/aiconfig/cls2n2jpk00pnpe1n5xx8uub2)! Make edits to the space and click **Share Workbook** to get a read-only view of your own.
+
+## Quickstart
+
+### 1. Create a new space with Gradio Workbooks
+
+- **Start with a template**. Duplicate the [Gradio Workbook Quickstart space](https://huggingface.co/spaces/lastmileai/gradio-workbook-template)
+- **Create from scratch**. Create a [new Gradio SDK space](https://huggingface.co/new-space) and add these files to your space repo. To add files to your space repo, you can do so through the [Web UI](https://huggingface.co/docs/hub/repositories-getting-started#adding-files-to-a-repository-web-ui) or [terminal](https://huggingface.co/docs/hub/repositories-getting-started#terminal).
+  - [app.py](https://huggingface.co/spaces/lastmileai/gradio-workbook-template/blob/main/app.py)
+  - [requirements.txt](https://huggingface.co/spaces/lastmileai/gradio-workbook-template/blob/main/requirements.txt)
+
+### 2. Design your space
+
+- Add models and customize. Edits/changes won’t save automatically.
+
+### 3. Lock in your edits
+
+- Click **Download**, which downloads the `aiconfig` file. This captures the current configuration of your space.
+- Rename the downloaded file to `my_app.aiconfig.json`.
+- Replace `my_app.aiconfig.json` in the space repo with the downloaded one.
+
+Anyone who visits your space will see the state represented by `my_app.aiconfig.json`.
+
+## Examples
+
+- [Music Playground](https://huggingface.co/spaces/lastmileai/music_generator). Make your demo more captivating with different modalities.
+- [AI Story Generator](https://huggingface.co/spaces/lastmileai/ai_story_generator). Easy to chain models for a workflow like story building.
+- [Stable Diffusion XL Playground](https://huggingface.co/spaces/lastmileai/sdxl_playground). Create interactive prompt templates for your model.
```

Comment:
"Create interactive prompt templates for your model with prompt and global parameters".

This space has no global params. We should have one cell that's called "using_global_paramaters" or something to use this as an example

Also not sure how to exactly make this clear, but this was a super big draw for NextEra. When they realized you can pass/override in your own params dynamically while calling `config.run()` instead of only relying on what's defined in the config, they got pretty excited. Ex: 
```
animals = ["horse", "rabbit", "duck", "dog"]
for animal in animals:
  await config.run("translate_animal_name", params={"animal": animal})
```

Reply:
It uses local parameters